protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/esmtp/MailSizeEsmtpExtension.java

```diff
@@ -148,16 +151,21 @@ private HookResult doMailSize(SMTPSession session,
      * @see org.apache.james.protocols.smtp.core.DataLineFilter#onLine(SMTPSession, byte[], LineHandler)
      */
     public Response onLine(SMTPSession session, ByteBuffer line, LineHandler<SMTPSession> next) {
-        Response response = null;
-    	Boolean failed = (Boolean) session.getAttachment(MESG_FAILED, State.Transaction);
+        Boolean failed = (Boolean) session.getAttachment(MESG_FAILED, State.Transaction);
         // If we already defined we failed and sent a reply we should simply
         // wait for a CRLF.CRLF to be sent by the client.
-        if (failed != null && failed.booleanValue()) {
-            // TODO
+        if (failed != null && failed) {
+            if (isDataTerminated(line)) {
+                line.rewind();
+                next.onLine(session, ByteBuffer.wrap(".\r\n".getBytes()));
```

Comment:
Yes, but it's a bug and we spotted it, it doesn't cost much at this point to fix it